### PR TITLE
Describe creating a socket-based service

### PR DESCRIPTION
**Pull request by pwmarcz**

### CODE REVIEW

File: developer/services/qrexec-socket-services.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+```
+sudo ln -s /lib/systemd/user/qubes-qrexec-policy-agent.socket /lib/systemd/user/sockets.target.wants/
+```
+
+### Link in qubes-rpc
+
+`qrexec-policy-agent` will handle a QubesRPC service called `policy.Ask`, so we add a link:
+
+```
+ln -s /etc/qubes-rpc/policy.Ask /var/run/qubes/policy-agent.sock
```
Comment (marmarek) [inline developer/services/qrexec-socket-services.md]: Wrong arg order.

File: developer/services/qrexec-socket-services.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        
+        # Process service_descriptor and input_data, prepare response.
+        ...
+        
+        writer.write(response)
+        await writer.drain()
+    finally:
+        writer.close()
+        await writer.wait_closed()
+```
```
Comment (marmarek) [inline developer/services/qrexec-socket-services.md]: Add example of calling run_server?

**Comment by marmarek** [comment]
Does the WIP prefix mean you plan to add something here?

**Comment by pwmarcz** [comment]
I'm done and I wanted to remove WIP prefix as soon as build passes. Right
now, I think, the only thing missing is the gbulb package for debian
stretch.

**Comment by pwmarcz** [comment]
Ah, sorry, wrong PR. This is waiting for the other PR just in case I modify
the code over there, because I might want to modify this one as well.

**Pull request merged**